Rust/src/imputevisitor.rs

```diff
@@ -59,25 +60,25 @@ impl Visitor<f64> for ImputeVisitor {
         } else {
             score = (self.score_unseen)(node_view.get_depth(), mass);
         }
-
+        let dist = new_point.iter().zip(leaf_point).map(|(a,b)| abs(a-b)).sum();
         self.values
-            .push((converged, score, node_view.get_leaf_index(), new_point));
+            .push((converged, score, node_view.get_leaf_index(), new_point, dist));
     }
 
     fn accept(&mut self, _point: &[f32], node_view: &dyn NodeView) {
-        let (converged, score, index, result) = self.values.pop().unwrap();
+        let (converged, score, index, result,dist) = self.values.pop().unwrap();
         if !converged {
             let prob = if self.ignore_mass == 0 {
                 node_view.get_probability_of_cut(&result)
             } else {
                 node_view.get_shadow_box().probability_of_cut(&result)
             };
             if prob == 0.0 {
-                self.values.push((true, score, index, result));
+                self.values.push((true, score, index, result,dist));
```

Comment:
Once we get to a 4-tuple it starts to look like a struct with named fields would be easier.

Reply:
Done